### PR TITLE
configures the storage layer to be backward compatible with OpenShift API

### PR DESCRIPTION
**Pull request by p0lyn0mial**
this PR adds default resources prefixes for `oauth ` and `user` groups to be backward compatible with OpenShift API.

It also sets the default storage media type to `application/vnd.kubernetes.protobuf` to be backward compatible with OpenShift API

### CODE REVIEW

File: pkg/oauth/apiserver/apiserver.go [hunk 1]
```diff
@@ -27,8 +25,8 @@ import (
 )
 
 var (
-	scheme = runtime.NewScheme()
-	codecs = serializer.NewCodecFactory(scheme)
+	scheme = legacyscheme.Scheme
```
Comment (p0lyn0mial) [inline pkg/oauth/apiserver/apiserver.go:28]: is it okay to refer to `legacyscheme.Scheme` ? Or do we prefer to plug a new one instead ?

File: pkg/cmd/oauth-apiserver/cmd.go [hunk 2]
```diff
@@ -111,5 +116,46 @@ func (o *OAuthAPIServerOptions) NewOAuthAPIServer() (*apiserver.OAuthAPIServer,
 		return nil, err
 	}
 
+	etcdOptions := o.RecommendedOptions.Etcd
+	etcdOptions.DefaultStorageMediaType = "application/vnd.kubernetes.protobuf"
```
Comment (p0lyn0mial) [inline pkg/cmd/oauth-apiserver/cmd.go:120]: this could be configured via the `storage-media-type` flag

File: pkg/cmd/oauth-apiserver/cmd.go [hunk 3]
```diff
@@ -111,5 +116,46 @@ func (o *OAuthAPIServerOptions) NewOAuthAPIServer() (*apiserver.OAuthAPIServer,
...
+	if err := o.RecommendedOptions.Etcd.ApplyWithStorageFactoryTo(storageFactory, &serverConfig.GenericConfig.Config); err != nil {
+		return nil, err
+	}
+
 	return serverConfig.Complete().New(genericapiserver.NewEmptyDelegate())
 }
+
+// specialDefaultResourcePrefixes are a custom storage prefixes (we must be backward compatible with OpenShift API)
+var specialDefaultResourcePrefixes = map[schema.GroupResource]string{
+	{Resource: "oauthaccesstokens"}:                                 "oauth/accesstokens",
```
Comment (sttts) [inline pkg/cmd/oauth-apiserver/cmd.go:149]: the non-grouped variants are probably not necessary anymore.

File: pkg/cmd/oauth-apiserver/cmd.go [hunk 4]
```diff
@@ -111,5 +116,46 @@ func (o *OAuthAPIServerOptions) NewOAuthAPIServer() (*apiserver.OAuthAPIServer,
...
+	{Resource: "oauthauthorizetokens"}:                              "oauth/authorizetokens",
+	{Resource: "oauthauthorizetokens", Group: "oauth.openshift.io"}: "oauth/authorizetokens",
+
+	{Resource: "oauthclients"}:                                           "oauth/clients",
+	{Resource: "oauthclients", Group: "oauth.openshift.io"}:              "oauth/clients",
+	{Resource: "oauthclientauthorizations"}:                              "oauth/clientauthorizations",
+	{Resource: "oauthclientauthorizations", Group: "oauth.openshift.io"}: "oauth/clientauthorizations",
+
+	{Resource: "identities"}:                             "useridentities",
+	{Resource: "identities", Group: "user.openshift.io"}: "useridentities",
```
Comment (sttts) [inline pkg/cmd/oauth-apiserver/cmd.go:160]: do we access these?

File: pkg/oauth/apiserver/apiserver.go (hunk 1 above)
Comment (sttts) [reply pkg/oauth/apiserver/apiserver.go:28]: custom

File: pkg/cmd/oauth-apiserver/cmd.go (hunk 4 above)
Comment (p0lyn0mial) [reply pkg/cmd/oauth-apiserver/cmd.go:160]: the new server will manage `user.openshift.io`

File: pkg/cmd/oauth-apiserver/cmd.go (hunk 3 above)
Comment (p0lyn0mial) [reply pkg/cmd/oauth-apiserver/cmd.go:149]: okay, will remove.

**Comment by p0lyn0mial** [comment]
/retitle configures the storage layer to be backward compatible with OpenShift API

File: pkg/cmd/oauth-apiserver/cmd.go (hunk 4 above)
Comment (sttts) [reply pkg/cmd/oauth-apiserver/cmd.go:152]: oh, right.

**Pull request merged**